crates/vision/src/image_segmenter.rs

```diff
@@ -886,28 +879,38 @@ mod tests {
 
     #[test]
     fn image_with_three_vertical_decreasing_segments_with_median() {
+        let row = |y| {
+            [
+                YCbCr422::new(y, 0, 0, 0),
+                YCbCr422::new(y, 0, 0, 0),
+                YCbCr422::new(y, 0, 0, 0),
+            ]
+        };
         let image = YCbCr422Image::from_ycbcr_buffer(
-            1,
+            3,
             12,
-            vec![
+            [
                 // only evaluating every secondth 422 pixel
-                YCbCr422::new(3, 0, 0, 0),
-                YCbCr422::new(3, 0, 0, 0), // skipped
-                YCbCr422::new(2, 0, 0, 0),
+                row(3),
+                row(3), // skipped
+                row(2),
```

Comment:
Why 9 lines with `YCbCr422::new(y, 0, 0, 0)` instead of three as previously?

Reply:
The test image used to be 1x12 and now is 3x12 to accommodate taking a horizontal median of three.
I'm not sure where you're getting the 9 from.

Reply:
Since row(y) now creates three values. And 3 * row = 3 * 3 = 9. But that makes sence since the image width is now three...